src/integrationTest/java/org/opensearch/test/framework/matcher/ClusterContainSuccessSnapshotMatcher.java

```diff
@@ -0,0 +1,67 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*
+*/
+package org.opensearch.test.framework.matcher;
+
+import java.util.stream.Collectors;
+
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import org.opensearch.action.admin.cluster.snapshots.get.GetSnapshotsRequest;
+import org.opensearch.action.admin.cluster.snapshots.get.GetSnapshotsResponse;
+import org.opensearch.client.Client;
+import org.opensearch.snapshots.SnapshotMissingException;
+import org.opensearch.snapshots.SnapshotState;
+import org.opensearch.test.framework.cluster.LocalCluster;
+
+import static java.util.Objects.requireNonNull;
+
+class ClusterContainSuccessSnapshotMatcher extends TypeSafeDiagnosingMatcher<LocalCluster> {
+
+	private final String repositoryName;
+	private final String snapshotName;
+
+	public ClusterContainSuccessSnapshotMatcher(String repositoryName, String snapshotName) {
+		this.repositoryName = requireNonNull(repositoryName, "Snapshot repository name is required.");
+		this.snapshotName = requireNonNull(snapshotName, "Snapshot name is required.");
+	}
+
+	@Override
+	protected boolean matchesSafely(LocalCluster cluster, Description mismatchDescription) {
```

Comment:
Rather than passing the local cluster, I think the interface would be much cleaner passing just the client to the cluster, this would also facilicate moving this code into a fully reusable test framework for all plugins - what do you think?

Reply:
done